ts/Stock/RangeSelector/RangeSelector.ts

```diff
@@ -2069,9 +2223,20 @@ class RangeSelector {
     ): void {
         const chart = this.chart;
         merge(true, chart.options.rangeSelector, options);
+        merge(true, this.options, options);
+        if (
+            this.options.selected &&
+            this.options.selected >= this.options.buttons!.length
+        ) {
+            this.options.selected = void 0;
+            chart.options.rangeSelector!.selected = void 0;
```

Comment:
With the change about `merge()` above, only `this.options.selected` could be changed. I see a potential bug with `selected = 0`, but that could be a misconfigured chart. Please make sure `update({ selected: 0, buttons : [] })` doesn't throw any errors